### DISCUSSION TITLE
Add support for `pixi` installations with `pixi.lock` files and `pyproject.toml` or `pixi.toml`

### DISCUSSION THREAD

**Discussion by benglewis** [aquasecurity/trivy#6860] [Ideas]
### Description

The new package manager in Python and Rust land with the name [`pixi`](https://prefix.dev/) is really fast, convenient (it has the optional to create feature sections, platform-specific dependencies and different environments, which can share dependency feature blocks, as well as automatic lock file maintenance, and most importantly, for us, support both for PyPI packages as well as Conda packages from `conda-forge` and other repositories like `nvidia` and `pytorch`).

Since it uses a different lock file than the existing packages, I believe that Trivy will not work as currently built. I may be wrong and I would be happy to discover that.

We are currently using [SafetyCLI](https://safetycli.com/) to scan the `pyproject.toml` for regular PyPI packages and [`jake`](https://github.com/sonatype-nexus-community/jake) to scan our Conda packages using `micromamba` to get the list of the package versions - yes it is a bit ugly.

### Target

Git Repository

### Scanner

Vulnerability

**Comment by nikpivkin** [comment]
Hi @benglewis !

Is `pixi` popular? I found <400 results using `pixi.lock` on [GitHub](https://github.com/search?q=path%3Apixi.lock&type=code).  And <600 for `pixi.toml` https://github.com/search?q=path%3Apixi.toml&type=code.

**Reply by marcelovilla** [reply]
Seems like both searches are over 2k results now. We're transitioning from pip/conda to uv/pixi in one of our projects and we'd love to see this happening :)

**Reply by ndeybach** [reply]
11.3K for pixi.toml and 7.6K for pixi.lock (and this later one was to be gitignore before recent 0.68.0 release of pixi, that now is all relative paths and fully commit compatible in pixi.lock v7).

It is absolutely being massively adopted in robotics, bioscience and cuda heavy stuff. It is also now the default toolchain used by conda-forge since last year and beginning to be mature enough to be an almost direct replacement / successor to conda by many aspects !

@nikpivkin , could this be reevaluated ?

**Comment by benglewis** [comment]
It is very new, so it isn’t extremely popular yet, but it is the best solution for the problems that it solves. There aren’t really any better solutions out there. I also imagine that those that need lock files are mostly not publishing to public GitHub repos

**Reply by johndutchover** [reply]
[Pixi ](https://github.com/prefix-dev/pixi)is growing fast and gaining popularity. [prefix.dev](https://github.com/prefix-dev) is doing some great work.

**Reply by nikpivkin** [reply]
@knqyf263 WDYT?

**Reply by knqyf263** [reply]
We'll wait for comments and upvotes from the community to see how much this feature is needed.

**Comment by aarz-snl** [comment]
Commenting -- wish we had this as well

**Comment by squinky86** [comment]
I've been testing the pixi support at https://github.com/squinky86/trivalire/tree/trixi as both Ada ALIRE (#11244) and Pixi are two of the key ecosystems being used for safety-critical, memory-safe software. Please let me know if the Trivy maintainers would welcome a pull request for this.
